src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs

```diff
@@ -78,6 +88,35 @@ await StoreWorkitemBlobAsync(identifier, dataset, cancellationToken)
             }
         }
 
+        /// <inheritdoc />
+        public async Task<QueryWorkitemResourceResponse> QueryAsync(
+            QueryParameters parameters,
+            CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(parameters);
+
+            var queryTags = await _workitemQueryTagService.GetQueryTagsAsync(cancellationToken: cancellationToken);
+
+            QueryExpression queryExpression = _queryParser.Parse(parameters, queryTags);
+
+            var partitionKey = _contextAccessor.RequestContext.GetPartitionKey();
+
+            WorkitemQueryResult queryResult = await _indexWorkitemStore.QueryAsync(partitionKey, queryExpression, cancellationToken);
+
+            if (!queryResult.WorkitemInstances.Any())
+            {
+                return new QueryWorkitemResourceResponse(Array.Empty<DicomDataset>(), queryExpression.ErroneousTags);
+            }
+
+            IEnumerable<DicomDataset> workitems = await Task.WhenAll(
```

Comment:
Is there limit on the queryResult? Please make sure we are not exposing unlimited download of any number of work items

Reply:
it is similar to QIDO. We only return 200